plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java

```diff
@@ -199,6 +217,31 @@ private MLUndeployModelNodeResponse createUndeployModelNodeResponse(MLUndeployMo
 
         Map<String, Integer> modelWorkerNodeCounts = new HashMap<>();
         boolean specifiedModelIds = modelIds != null && modelIds.length > 0;
+
+        Set<String> invalidAccessModels = new HashSet<>();
+        User user = RestActionUtils.getUserContext(client);
```

Comment:
As we discussed offline, we can't get user information in this class. I debugged by adding a new transport action similar to deploy model, we can get user information. You can refer to my code https://github.com/ylwu-amzn/ml-commons/commit/862ad390ca9624a3e11a3c2d38974054e8482fc9, you can get user information and verify model group backend role in this class `TransportUndeployModelsAction`.

Reply:
Thanks Yaliang. I think this should work then. Will add these changes.